packages/formdescriptioneditors/frontend/sirius-components-formdescriptioneditors/src/CheckboxWidget.tsx

```diff
@@ -14,32 +14,53 @@ import { getCSSColor, useSelection } from '@eclipse-sirius/sirius-components-cor
 import { CheckboxStyleProps } from '@eclipse-sirius/sirius-components-forms';
 import HelpOutlineOutlined from '@mui/icons-material/HelpOutlineOutlined';
 import Checkbox from '@mui/material/Checkbox';
-import FormControlLabel from '@mui/material/FormControlLabel';
+import FormControl from '@mui/material/FormControl';
 import Typography from '@mui/material/Typography';
 import { useEffect, useRef, useState } from 'react';
 import { makeStyles } from 'tss-react/mui';
 import { CheckboxWidgetProps } from './WidgetEntry.types';
 
-const useStyles = makeStyles<CheckboxStyleProps>()((theme, { color }) => ({
-  style: {
-    color: color ? getCSSColor(color, theme) : theme.palette.primary.light,
-    '&.Mui-checked': {
+const useStyles = makeStyles<CheckboxStyleProps>()((theme, { color, gridProps }) => {
+  const { gridTemplateColumns, gridTemplateRows, labelGridColumn, labelGridRow, widgetGridColumn, widgetGridRow, gap } =
+    {
+      ...gridProps,
+    };
+  return {
+    style: {
       color: color ? getCSSColor(color, theme) : theme.palette.primary.light,
+      '&.Mui-checked': {
+        color: color ? getCSSColor(color, theme) : theme.palette.primary.light,
+      },
     },
-  },
-  selected: {
-    color: theme.palette.primary.main,
-  },
-  propertySectionLabel: {
-    display: 'flex',
-    flexDirection: 'row',
-    alignItems: 'center',
-  },
-}));
+    selected: {
+      color: theme.palette.primary.main,
+    },
+    propertySection: {
+      display: 'grid',
+      gridTemplateColumns: gridTemplateColumns ?? 'min-content 1fr',
+      gridTemplateRows,
+      alignItems: 'center',
+      gap: gap ?? '',
+    },
+    propertySectionLabel: {
+      gridColumn: labelGridColumn ?? '2/3',
+      gridRow: labelGridRow,
+      display: 'flex',
+      flexDirection: 'row',
+      alignItems: 'center',
+      justifyContent: 'space-between',
+    },
+    propertySectionWidget: {
+      gridColumn: widgetGridColumn,
+      gridRow: widgetGridRow ?? '1/2',
```

Comment:
gridColumn and gridRow are used for both label and widget div so we need to distinguish the names

Reply:
not sure to understand the question.
I can add gridColumn default value with '1/2' but gridRow is also needed otherwise the div is located as if gridRow  ware '2/3'

Reply:
Ok, it's due to the fact that by default, you need to reverse the order of the label and widget.